Assets/_Functionalities/Wms/Scripts/BoundingBox.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using Netherlands3D.Coordinates;
+
+namespace Netherlands3D.Twin.Wms
+{
+    public class BoundingBox
+    {
+        public Coordinate BottomLeft { get; private set; }
+        public Coordinate TopRight { get; private set; }
+        public CoordinateSystem CoordinateSystem { get; private set; }
+
+        public BoundingBox(Coordinate bottomLeft, Coordinate topRight)
+        {
+            if (bottomLeft.Points[0] > topRight.Points[0] || bottomLeft.Points[1] > topRight.Points[1])
+            {
+                throw new ArgumentException(
+                    "Invalid coordinates for BoundingBox. BottomLeft should have lower values than TopRight."
+                );
+            }
+
+            if (topRight.CoordinateSystem != bottomLeft.CoordinateSystem)
+            {
+                topRight = topRight.Convert((CoordinateSystem)bottomLeft.CoordinateSystem);
+            }
+
+            BottomLeft = bottomLeft;
+            TopRight = topRight;
+            CoordinateSystem = (CoordinateSystem)bottomLeft.CoordinateSystem;
+        }
+
+        public void Convert(CoordinateSystem coordinateSystem)
+        {
+            BottomLeft = BottomLeft.Convert(coordinateSystem);
+            TopRight = TopRight.Convert(coordinateSystem);
+        }
+
+        // Method to check if this BoundingBox contains another BoundingBox completely
+        public bool Contains(BoundingBox other)
+        {
+            return other.BottomLeft.Points[0] >= this.BottomLeft.Points[0] &&
+                   other.BottomLeft.Points[1] >= this.BottomLeft.Points[1] &&
```

Comment:
is het niet handiger hier om een Bounds class te gebruiken die heeft dit soort checks standaard

Reply:
Umm. Bas .. dit is een Boundingbox class speciaal voor dit soort checks 🤣